pkg/apis/networking/v1/defaults_test.go

```diff
@@ -234,6 +237,99 @@ func TestSetDefaultNetworkPolicy(t *testing.T) {
 	}
 }
 
+func TestSetDefaultsForIngressClassParametersReference(t *testing.T) {
+	tests := []struct {
+		original          *networkingv1.IngressClass
+		expected          *networkingv1.IngressClass
+		enableFeatureGate bool
+	}{
+		{ // Populated parameters should set the default Scope when feature is enabled
+			original: &networkingv1.IngressClass{
+				Spec: networkingv1.IngressClassSpec{
+					Controller: "controller",
+					Parameters: &networkingv1.IngressClassParametersReference{
+						Kind: "k",
+						Name: "n",
+					},
+				},
+			},
+			expected: &networkingv1.IngressClass{
+				Spec: networkingv1.IngressClassSpec{
+					Controller: "controller",
+					Parameters: &networkingv1.IngressClassParametersReference{
+						Kind:  "k",
+						Name:  "n",
+						Scope: networkingv1.IngressClassParametersReferenceScopeCluster,
+					},
+				},
+			},
+			enableFeatureGate: true,
+		},
+		{ // Empty Parameters should not set the default Scope when feature is enabled
+			original: &networkingv1.IngressClass{
+				Spec: networkingv1.IngressClassSpec{
+					Controller: "controller",
+				},
+			},
+			expected: &networkingv1.IngressClass{
+				Spec: networkingv1.IngressClassSpec{
+					Controller: "controller",
+				},
+			},
+			enableFeatureGate: true,
+		},
+		{ // Populated parameters should not set Scope when feature is disabled
+			original: &networkingv1.IngressClass{
+				Spec: networkingv1.IngressClassSpec{
+					Controller: "controller",
+					Parameters: &networkingv1.IngressClassParametersReference{
+						Kind: "k",
+						Name: "n",
+					},
+				},
+			},
+			expected: &networkingv1.IngressClass{
+				Spec: networkingv1.IngressClassSpec{
+					Controller: "controller",
+					Parameters: &networkingv1.IngressClassParametersReference{
+						Kind: "k",
+						Name: "n",
+					},
+				},
+			},
+			enableFeatureGate: false,
+		},
+		{ // Empty Parameters should not set the default Scope when feature is disabled
+			original: &networkingv1.IngressClass{
+				Spec: networkingv1.IngressClassSpec{
+					Controller: "controller",
+				},
+			},
+			expected: &networkingv1.IngressClass{
+				Spec: networkingv1.IngressClassSpec{
+					Controller: "controller",
+				},
+			},
+			enableFeatureGate: false,
+		},
+	}
+
+	for i, test := range tests {
+		original := test.original
+		expected := test.expected
+		defer featuregatetesting.SetFeatureGateDuringTest(t, utilfeature.DefaultFeatureGate, features.IngressClassNamespacedParams, test.enableFeatureGate)()
+		obj2 := roundTrip(t, runtime.Object(original))
+		got, ok := obj2.(*networkingv1.IngressClass)
+		if !ok {
+			t.Errorf("(%d) unexpected object: %v", i, got)
```

Comment:
Instead of `(%d)` in logs use `t.Run(test.name, ...` like https://github.com/kubernetes/kubernetes/blob/master/pkg/controller/endpointslice/endpointslice_controller_test.go#L1390.